src/HappyConnOpener.h

```diff
@@ -77,6 +77,8 @@ class HappyConnOpenerAnswer
     /// whether HappyConnOpener succeeded, returning a usable connection
     bool success() const { return !error; }
 
+    /// a candidate connection taken from 'destinations'
+    ResolvedPeer candidateConn;
     /// on success: an open, ready-to-use Squid-to-peer connection
     /// on failure: either a closed failed Squid-to-peer connection or nil
     Comm::ConnectionPointer conn;
```

Comment:
I counted at least two existing places where we already store ResolvedPeer and the corresponding ConnectionPointer. Addressing the "fresh Connection objects" change request will add one more such place -- the Attempt class. The two stored pieces of information are naturally closely linked. We should add a class to encapsulate those two members and represent their special relationship and/or reuse ResolvedPeer for that purpose. I suggest trying the latter first:

```suggestion
    /// on success: an open, ready-to-use Squid-to-peer connection (with a return receipt)
    /// on failure: either a closed failed Squid-to-peer connection or nil (w/o a return receipt)
    ResolvedPeer conn;
```

Similar changes should be done in other places that store both the return receipt and the connection. If it is necessary, we can reduce the number of changes by adding new reference members (with the old names) to refer to the Connection member inside ResolvedPeer, but I hope it will not be necessary.

If we go down this path, then the change request about renaming ResolvedPeer::connection() to address() should be interpreted as "add an address() method". Both methods would return the same thing but their users will be able to emphasize what they expect that "thing" to be.